forgerock-authenticator/src/main/java/org/forgerock/android/auth/AuthenticatorManager.java

```diff
@@ -240,9 +289,69 @@ List<PushNotification> getAllNotifications(@NonNull Mechanism mechanism) {
         }
     }
 
+    private void createCombinedMechanismsFromUri(String uri, FRAListener<Mechanism> listener) {
+        final Mechanism[] oathMechanism = new Mechanism[1];
+
+        // Check if security policies are available in the URI before proceed with registration
+        Logger.debug(TAG, "Evaluating policies for the new Account");
+        boolean policiesCompliance = policyEvaluator.evaluate(context, uri);
+        if (!policiesCompliance) {
+            listener.onException(new MechanismCreationException("This account cannot be " +
+                    "registered on this device. It violates the following policy: " +
+                    policyEvaluator.getNonCompliancePolicy().getName()));
+            return;
+        } else {
+            Logger.debug(TAG, "All policies passed for the new Account");
+        }
+
+        // Register OATH mechanism passing combined parameters
+        oathFactory.createFromUri(uri, new FRAListener<Mechanism>() {
+            @Override
+            public void onSuccess(Mechanism result) {
+                oathMechanism[0] = result;
+                Logger.debug(TAG, "OATH mechanism in Combined MFA successfully created.");
+            }
+
+            @Override
+            public void onException(Exception e) {
+                Logger.error(TAG, "Error creating OATH mechanism in Combined MFA.");
+            }
+        });
+
+        // Register PUSH mechanism passing combined parameters
+        pushFactory.createFromUri(uri, new FRAListener<Mechanism>() {
+            @Override
+            public void onSuccess(Mechanism result) {
+                listener.onSuccess(result);
+                Logger.debug(TAG, "PUSH mechanism in Combined MFA successfully created.");
+            }
+
+            @Override
+            public void onException(Exception e) {
+                listener.onSuccess(oathMechanism[0]);
+                listener.onException(e);
+                Logger.error(TAG, "Error creating PUSH mechanism in Combined MFA.");
+            }
+        });
+    }
+    
     private void initializeAccount(Account account) {
         if(account != null) {
-            Logger.debug(TAG, "Loading associated data for the Account with ID: %s", account.getId());
+            Logger.debug(TAG, "Evaluating policies for the new Account with ID: %s",
+                    account.getId());
+            boolean policiesCompliance = policyEvaluator.evaluate(context, account);
+            if(!account.isLocked() & !policiesCompliance) {
+                Logger.debug(TAG, "Lock Account due non-compliance policy with name: %s",
+                        policyEvaluator.getNonCompliancePolicy().getName());
+                account.lock(policyEvaluator.getNonCompliancePolicy());
```

Comment:
Looks like a bit danger to maintain state in the policyEvaluator, if calling the initializeAccount with a different Account, keeping the nonCompliancePolicy may be an issue.

Reply:
I will remove nonCompliancePolicy attribute. Developers will have to rely on the logs or the Account.getLockingPolicy

Reply:
Done, created a Result object to replace boolean as result